### PR TITLE
RUN-3721 Fixed issue where we expected an array for preload processing.

### PR DESCRIPTION
**Pull request by rdepena**
Test results: 

[Win7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5a3ac5156f202a5432b79f7c)
[Win10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5a3ac5986f202a5432b79f7f)

### CODE REVIEW

File: src/browser/preload_scripts.ts [hunk 1]
```diff
@@ -81,7 +81,7 @@ function downloadScript(identity: Identity, preloadScript: PreloadScript): Promi
 export async function loadScripts(identity: Identity): Promise<PreloadScriptWithContent[]|any> {
     const options = Window.getOptions(identity);
     const { preloadScripts } = convertToElectron(options);
-    const promises = preloadScripts.map((preloadScript: PreloadScript) => loadScript(identity, preloadScript));
+    const promises = preloadScripts ? preloadScripts.map((preloadScript: PreloadScript) => loadScript(identity, preloadScript)) : [];
```
Comment (HarsimranSingh) [inline src/browser/preload_scripts.ts]: I think the problem might lie earlier. Why doesn't this catch it https://github.com/HadoukenIO/core/blob/develop/src/browser/convert_options.js#L348

Comment (whyn07m3) [reply src/browser/preload_scripts.ts]: IMO, we should either fix this in [here](https://github.com/HadoukenIO/core/blob/develop/src/browser/convert_options.js#L258) instead
--- or ---
do this: `const { preloadScripts = [] } = convertToElectron(options);`

Comment (rdepena) [reply src/browser/preload_scripts.ts]: @whyn07m3 will make that change.

**Comment by rdepena** [comment]
@HarsimranSingh @whyn07m3 avoided changes to convert_options as they will affect preload script inheritance, a bit more changes than I want for a hotfix.

Test results: 
[Win7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5a53cf83bd581266877d664e)
[Win10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5a53d05dbd581266877d664f)

File: src/browser/preload_scripts.ts [hunk 2]
```diff
@@ -80,8 +80,8 @@ function downloadScript(identity: Identity, preloadScript: PreloadScript): Promi
 
 export async function loadScripts(identity: Identity): Promise<PreloadScriptWithContent[]|any> {
     const options = Window.getOptions(identity);
-    const { preloadScripts } = convertToElectron(options);
-    const promises = preloadScripts.map((preloadScript: PreloadScript) => loadScript(identity, preloadScript));
+    const { preloadScripts = [] }: {preloadScripts: Array<PreloadScript>} = convertToElectron(options);
```
Comment (whyn07m3) [inline src/browser/preload_scripts.ts]: Shouldn't this be `const { preloadScripts = [] }: PreloadScript[] = convertToElectron(options);`?

File: src/browser/preload_scripts.ts [hunk 3]
```diff
@@ -80,8 +80,8 @@ function downloadScript(identity: Identity, preloadScript: PreloadScript): Promi
...
+    const promises =  preloadScripts.map((preloadScript: PreloadScript) => loadScript(identity, preloadScript));
```
Comment (whyn07m3) [inline src/browser/preload_scripts.ts]: If you are going to change the above line, this line has an extra unnecessary space added and shows as a changed line

File: src/browser/preload_scripts.ts (hunk 2 above)
Comment (rdepena) [reply src/browser/preload_scripts.ts]: I actually prefer Array<T>

Comment (whyn07m3) [reply src/browser/preload_scripts.ts]: But I think the type is wrong though or am I missing something?
`const { preloadScripts = [] }: { preloadScripts: Array<PreloadScript> } = convertToElectron(options);`
I think it should be:
`const { preloadScripts = [] }: Array<PreloadScript> = convertToElectron(options);`
as resulting `preloadScripts` is an array, not an object with a property...

Comment (rdepena) [reply src/browser/preload_scripts.ts]: ahh, that is the correct syntax for destructing.

Comment (rdepena) [reply src/browser/preload_scripts.ts]: meaning the code there is correct, preloadScripts will be an array of PreloadScript.

Comment (rdepena) [reply src/browser/preload_scripts.ts]: https://www.typescriptlang.org/docs/handbook/variable-declarations.html#destructuring

**Pull request merged**